src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,74 @@
+        private readonly ITransactionReceiptRepository _receipts;
+        private readonly ITransactionReceivedEvent _transactionReceived;
+        private readonly IHashProvider _hashProvider;
+        public const string ComponentName = nameof(Web3EthApi);
+
+        public Web3EthApi(IStateReader stateReader, IDeltaResolver deltaResolver, IDeltaCache deltaCache, IDeltaExecutor executor, [KeyFilter(ComponentName)] IStorageProvider storageProvider, [KeyFilter(ComponentName)] IStateProvider stateProvider, ITransactionReceiptRepository receipts, ITransactionReceivedEvent transactionReceived, IHashProvider hashProvider)
         {
-            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
+            _receipts = receipts;
+            _transactionReceived = transactionReceived ?? throw new ArgumentNullException(nameof(transactionReceived));
+            _hashProvider = hashProvider;
             StateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
-            StateRootResolver = stateRootResolver ?? throw new ArgumentNullException(nameof(stateRootResolver));
             DeltaResolver = deltaResolver ?? throw new ArgumentNullException(nameof(deltaResolver));
+            DeltaCache = deltaCache ?? throw new ArgumentNullException(nameof(deltaCache));
+            Executor = executor ?? throw new ArgumentNullException(nameof(executor));
+            StorageProvider = storageProvider ?? throw new ArgumentNullException(nameof(storageProvider));
+            StateProvider = stateProvider ?? throw new ArgumentNullException(nameof(stateProvider));
         }
-        
-        public IDeltaExecutor DeltaExecutor { get; }
+
         public IStateReader StateReader { get; }
         public IDeltaResolver DeltaResolver { get; }
-        public IStateRootResolver StateRootResolver { get; }
+        public IDeltaCache DeltaCache { get; }
+
+        public IDeltaExecutor Executor { get; }
+        public IStorageProvider StorageProvider { get; }
+        public IStateProvider StateProvider { get; }
+        public ITransactionReceiptResolver ReceiptResolver => this;
+
+        public Keccak SendTransaction(PublicEntry publicEntry)
+        {
+            TransactionBroadcast broadcast = new TransactionBroadcast
+            {
+                PublicEntry = publicEntry
+            };
+
+            _transactionReceived.OnTransactionReceived(broadcast.ToProtocolMessage(new PeerId(), new CorrelationId(Guid.NewGuid())));
+
+            return new Keccak(_hashProvider.ComputeMultiHash(broadcast).Digest);
+        }
+
+        public TransactionReceipt Find(Keccak hash)
+        {
+            if (_receipts.TryFind(hash, out TransactionReceipt receipt))
+            {
+                return receipt;
+            }
+
+            throw new Exception("Receipt not found");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)